packages/bp/src/migrations/v12_24_0-1626065172-remove_global_channel_configs.ts

```diff
@@ -0,0 +1,33 @@
+import * as sdk from 'botpress/sdk'
+import { Migration, MigrationOpts } from 'core/migration'
+import _ from 'lodash'
+
+const ROOT_FOLDER = './config'
+const CHANNELS = ['messenger', 'slack', 'smooch', 'teams', 'telegram', 'twilio', 'vonage'] as const
+type Channels = typeof CHANNELS[number]
+
+const migration: Migration = {
+  info: {
+    description: 'Remove all global configs channel related',
+    target: 'core',
+    type: 'config'
+  },
+  up: async ({ bp }: MigrationOpts): Promise<sdk.MigrationResult> => {
+    const channelConfigsFilePath = await bp.ghost.forGlobal().directoryListing(ROOT_FOLDER, 'channel-*.json')
+    for (const file of channelConfigsFilePath) {
+      const channelName = file.replace('.json', '').replace('channel-', '')
+      if (!CHANNELS.includes(channelName as Channels)) {
+        continue
+      }
+
+      await bp.ghost.forGlobal().deleteFile(ROOT_FOLDER, file)
+    }
+
+    return { success: true, message: 'Configurations removed successfully' }
+  },
+  down: async (): Promise<sdk.MigrationResult> => {
+    return { success: true, message: 'Nothing to do skipping.' }
+  }
```

Comment:
The goal of up/down migration is to be able to revert to a working state, which isn't the case here since global configs would be gone. 

What do you think of saving those files elsewhere when migrating, so we can restore them if we need to downgrade? For example, we could move them from `global/config` to `global/backup/config_12_23_0`. 

I think we need a way to save "unrecoverable" data when migrating so the up/down works as expected. @EFF  your thoughts?

Reply:
> The goal of up/down migration is to be able to revert to a working state, which isn't the case here since global configs would be gone.

You are still in a working state since the global configurations are only there to display a template of the config each bots should implement. Furthermore, if you start Botpress and those files are missing they are automatically copied from the channel's config. Meaning that it would be useless to create a backup of those files.